Dockerfile

```diff
@@ -8,11 +8,15 @@ ENV \
 WORKDIR /usr/src
 
 ## Setup Home Assistant
+COPY requirements.txt requirements_all.txt homeassistant/
+COPY homeassistant/package_constraints.txt homeassistant/homeassistant/
+RUN \
+    pip3 install --no-cache-dir --no-index --only-binary=:all: --find-links "${WHEELS_LINKS}" \
+    -r homeassistant/requirements_all.txt
```

Comment:
We could make it 3 steps...

- `requirements.txt` contains what core minimally needs
- Install core
- install `requirements_all.txt`

The latter changes the most

Reply:
That particular order defeats the purpose somewhat: The idea is to avoid installing requirements again when I just made some code changes.

We could still split it up, it would speed up build when just requirements_all.txt changes, but not sure if that is worth it?

- install requirements.txt
- install requirements_all.txt
- Install core

Reply:
> The idea is to avoid installing requirements again when I just made some code changes.

That is why I respond on this, as `requirements_all` and the codebase change a lot (I'm pretty confident that the amount of builds without a dep bump is nihil).

Hence the original suggested change in this PR doesn't change a thing.

If anything, installing just `requirements.txt` would help, but should not have a big impact.

Reply:
>  (I'm pretty confident that the amount of builds without a dep bump is nihil).
> 
> Hence the original suggested change in this PR doesn't change a thing.

It does for me: My iteration time to build a new Core container (to test it with Supervisor and the whole stack) goes down significantly.

I guess we are discussing different use cases. My use case is local development, where I mostly change source code and not touch `requirements.txt` or `requirements_all.txt` that often.

Reply:
> It does for me: My iteration time to build a new Core container (to test it with Supervisor and the whole stack) goes down significantly.

Ah right, yeah in that case:

- install requirements.txt
- install requirements_all.txt
- Install core

Is the better order :)

Reply:
We should update the `Dockerfile.dev` in that case as well.

Reply:
Hm, seems we don't copy source code there, I guess its bind mounted by VS Code to `/workspaces`.

Reply:
Ah, did you meant separating requirements installation? In `Dockerfile.dev` we don't install `requirements_all.txt`, but we do install `requirements_test.txt`. We could separate those into two RUN commands to increase caching opportunities.

Reply:
Updated.